hashintel/hash
# Hotfix comments

## 🌟 What is the purpose of this PR?

While moving to multi-axis temporal versioning, we accidentally broke comments. This wasn't picked up as we haven't addressed our test deficiency yet, and we were manually specifying the type of the GraphQL resolver rather than relying on generated types (which meant TypeScript was effectively _told_ the type was a `string` when it was not).

This PR addresses the above.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202805690238892/1203756332746894/f) _(internal)_

## 🚫 Blocked by

N/A

## 🔍 What does this change?

- See commits, diff is small

## 📜 Does this require a change to the docs?

Not to my knowledge

## ⚠️ Known issues

- A lot of our GraphQL definitions have the same problem identified as part of this PR. This is tracked in this [Asana task](https://app.asana.com/0/1202805690238892/1203749177059860/f) _(internal)_
- Some of the properties on the `Comment` _entity_ are still stringified timestamps, this has been identified in the code and captured in an [Asana task](https://app.asana.com/0/1202805690238892/1203763454493756/f) _(internal)_

## 🐾 Next steps

- Update the other GraphQL definitions to use generated types and catch errors like this
- Decide on how to handle the timestamps in `Comment`

## 🛡 What tests cover this?

To my knowledge, none outside of linting, hence the issue. We will be addressing testing soon, so unfortunately this PR does not address that at the same time.

## ❓ How to test this?

1.  Checkout the branch / view the deployment
2.  Run the app
3.  Try creating a comment and ensure the app does not crash

## 📹 Demo

Left as an exercise to the reader
